src/redux/saga/addFeed.js

```diff
@@ -0,0 +1,37 @@
+import { call, put, takeEvery } from "redux-saga/effects";
+
+
+import { setCategories, 
+         setTotalCategoriesCount, 
+         setCurrentCategoriesPage, 
+        } from "../actions/addFeed";
+
+import rssResponseParser from "../../helpers/parsers/rss-response-parser";
+import getFeedAPI from './../../api'
+import { FEED_FETCH_REQUESTED, CATEGORY_PAGE_CHANGED } from "../actions/types";
+
+function* fetchFeed(action) {
+  try {
+    const xmlText = yield call(getFeedAPI, action.url);
+    const feedObj = yield rssResponseParser(xmlText);
+    yield put(setCategories(feedObj.feedTitle, feedObj.feedLink, feedObj.feedCategories, feedObj.feedImage));
```

Comment:
can we destructure all these variables from `feedObj` not to write `feedObj.` all the time?
const {feedTitle, feedLink, etc. } = feedObj;